spec/controllers/notes_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+RSpec.describe NotesController, type: :controller do
+  describe "GET index" do
+    let(:user) { FactoryBot.create(:user) }
+    let(:note1) { FactoryBot.create(:note, user: user) }
+    let(:note2) { FactoryBot.create(:note, user: user, title: "New note") }
+    let(:note3) { FactoryBot.create(:note, user: user, title: "Ruby is awesome") }
+
+    before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
```

Comment:
What does this line do?

Reply:
This is to bring all routes related to devise user to the `@request`'s  environment during testing.